tests/ulyanov_et_al_2016/test_modules.py

```diff
@@ -0,0 +1,143 @@
+    with subtests.test("kernel_size"):
+        assert module.kernel_size == pytest.approx(2)
+    with subtests.test("stride"):
+        assert module.stride == pytest.approx(2)
+    with subtests.test("padding"):
+        assert module.padding == pytest.approx(0)
+
+
+def test_UlyanovEtAl2016TextureDownsample(mocker, input_image):
+    mock = mocker.patch(
+        "pystiche_papers.ulyanov_et_al_2016.modules.TextureNoiseParams.downsample"
+    )
+    module = modules.UlyanovEtAl2016TextureDownsample()
+    module(input_image)
+    mock.assert_called_once()
+
+
+def test_ulyanov_et_al_2016_downsample(subtests):
+    for stylization in (True, False):
+        with subtests.test(stylization=stylization):
+            module = modules.ulyanov_et_al_2016_downsample(stylization=stylization)
+            assert isinstance(
+                module,
+                modules.UlyanovEtAl2016StylizationDownsample
+                if stylization
+                else modules.UlyanovEtAl2016TextureDownsample,
+            )
+
+
+def test_ulyanov_et_al_2016_upsample(subtests):
+    module = modules.ulyanov_et_al_2016_upsample()
+    assert isinstance(module, nn.Upsample)
+    with subtests.test("scale_factor"):
+        assert module.scale_factor == pytest.approx(2.0)
+    with subtests.test("mode"):
+        assert module.mode == "nearest"
+
+
+def test_get_norm_module(subtests):
+    in_channels = 3
+    for instance_norm in (True, False):
+        with subtests.test(instance_norm=instance_norm):
+            norm_module = modules.get_norm_module(
+                in_channels, instance_norm=instance_norm
+            )
+
+            assert isinstance(
+                type(norm_module),
+                type(nn.InstanceNorm2d) if instance_norm else type(nn.BatchNorm2d),
+            )
+
+            with subtests.test("out_channels"):
+                assert norm_module.num_features == in_channels
+
+            with subtests.test("eps"):
+                assert norm_module.eps == pytest.approx(1e-5)
+
+            with subtests.test("momentum"):
+                assert norm_module.momentum == pytest.approx(1e-1)
+
+            with subtests.test("affine"):
+                assert norm_module.affine
+
+            with subtests.test("track_running_stats"):
+                assert norm_module.track_running_stats
+
+
+def test_get_activation_module(subtests):
+    for instance_norm, impl_params in itertools.product((True, False), (True, False)):
+        with subtests.test(instance_norm=instance_norm):
+            norm_module = modules.get_activation_module(
+                impl_params=impl_params, instance_norm=instance_norm
+            )
+
+            assert isinstance(
+                type(norm_module),
+                type(nn.ReLU) if impl_params and instance_norm else type(nn.LeakyReLU),
+            )
+
+            with subtests.test("inplace"):
+                assert norm_module.inplace
+
+            if not impl_params and instance_norm:
```

Comment:
At this point we have already checked if we only get an `nn.LeakyReLU` for the correct parameter combination.

```suggestion
            if isinstance(norm_module, nn.LeakyReLU):
```